devito/compiler.py

```diff
@@ -152,12 +154,33 @@ def __init__(self, *args, **kwargs):
             log("WARNING: Running on Intel KNL without OpenMP is highly discouraged")
 
 
+class CustomCompiler(Compiler):
+    """Custom compiler based on standard environment flags
+
+    :param openmp: Boolean indicating if openmp is enabled. False by default
+
+    Note: Currently honours CC, CFLAGS, LD and LDFLAGS, with defaults similar
+    to the default GNU settings.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(CustomCompiler, self).__init__(*args, **kwargs)
+        self.cc = environ.get('CC', 'g++')
+        self.ld = environ.get('LD', 'g++')
+        self.cflags = environ.get('CFLAGS', '-O3 -g -fPIC -Wall').split(' ')
+        self.ldflags = environ.get('LDFLAGS', '-shared').split(' ')
+
+
 # Registry dict for deriving Compiler classes according to
 # environment variable DEVITO_ARCH. Developers should add
 # new compiler classes here and provide a description in
 # the docstring of get_compiler_from_env().
 compiler_registry = {
     'gcc': GNUCompiler, 'gnu': GNUCompiler,
+    'gcc-4.9': partial(GNUCompiler, version='4.9'),
+    'g++-4.9': partial(GNUCompiler, version='4.9'),
+    'gcc-5': partial(GNUCompiler, version='5'),
+    'g++-5': partial(GNUCompiler, version='5'),
     'clang': ClangCompiler, 'osx': ClangCompiler,
```

Comment:
I really don't think there are any standard names here.
Anything beyond g++ and gcc should go into CC and CXX.